Add /execs to other Notebooks #322

RavenMarQ opened the pull request:
PR for all execs to be made from notebooks

RavenMarQ committed:
Test

This is to test Committing and PR

RavenMarQ committed:
A untested prototype of understand_parse_dependencies

Modeled after parse_dependencies with a similar output, this is for review to see if it is in-line with specification

RavenMarQ committed:
Completed Functions

Per the specifications in Issue 308, the files are functional and ready to be put into a notebook.

RavenMarQ committed:
Update src.R

Quick fix

RavenMarQ committed:
Working On Creating Notebook

Updating NAMESPACE to export the new functions and creating the Rmd, are the primary notes. The folder holding the sample project also is included locally, but uses the calculator project provided in Issue from Carlos

RavenMarQ committed:
Addressing the Code Review

Most of the code review notes have been resolved, except for changing the file.path and descriptions. The notebook is currently being updated, but a preview on the proposed format is provided.

RavenMarQ committed:
About-Completed Milestone

Having addressing most things, especially using file.path, the functions are functional and the notebook works.

RavenMarQ committed:
Final Touches

Resolving comments received on the notebook.

beydlern committed:
i #308 Refactored understand_showcase.Rmd

- Refactored the understand_showcase.Rmd notebook to expect the use of the getters from R/config.R (i #230 contains the getter functions in R/config.R).

RavenMarQ committed:
Merge branch 'master' into 308-scitool-understand-parse

RavenMarQ committed:
After Merge with master Branch

RavenMarQ committed:
Documentation for the three functions

RavenMarQ committed:
Fixing mismatched name in Documentation

beydlern committed:
i #308 Fixed Relative Paths in a Notebook

- The project configuration sections of a notebook was incorrectly using the project directory (kaiaulu/) as its working directory rather than the directory that it resides in (/vignettes/) as its working directory.

carlosparadis committed:
Merge branch 'master' into 308-scitool-understand-parse

RavenMarQ committed:
Create Understand Notebook Exec

Currently non-functional, but a push for code review

daomcgill commented on exec/understand.R:
```diff
@@ -0,0 +1,57 @@
+  if (arguments[["--file"]]) {
+    result <- parse_understand_dependencies(understand_dir = understand_folder, parse_type = "file")
+  } else if (arguments[["--class"]]) {
+    result <- parse_understand_dependencies(understand_dir = understand_folder, parse_type = "classs")
```
typo

daomcgill commented on exec/understand.R:
```diff
@@ -0,0 +1,57 @@
+} else {
+  stop("No/invalid option(s) provided.")
+}
+
```
Write the output to a file before the last else

daomcgill reviewed (commented):
I left my feedback, though it may be best if @carlosparadis puts eyes on this before you make the changes.

The flow of this script makes it difficult to read. I think enough so that you would want to restructure this. Take a look at the other exec scripts. The logic should look something like this:
```
1. if "build" and "help":
- useful message to describe build

2. else if "build":
- load config file from args
- use getters to set params
- call the function
- write output to file

3. if "parse" and "help":
- useful message to describe build

4. else if "parse" and file:
- load config file from args
- use getters to set params
- call the function
- write output to file

5. same thing with parse and class
```

See inline for my other comments. Notably: do not check for dir exists in here as it will be error-prone. Mark is working on making sure the directory will exist.

Oh, also do not forget to merge the necessary functions for testing/running this into the branch.

RavenMarQ committed:
Merge branch '308-scitool-understand-parse' into 322-creating-exec-for-notebooks

RavenMarQ committed:
Update understand.R

From my understanding of the feedback received, updated the understand.R.

RavenMarQ committed:
Uploaded Completed executables

Simple push after viewing during 12/5/2024 meeting

carlosparadis commented on .idea/.gitignore:
same issue as M1: Please remove the .idea folder.

carlosparadis commented on .idea/kaiaulu.iml:
Here again and below.

carlosparadis commented on exec/understand.R:
```diff
@@ -0,0 +1,68 @@
+  understand_folder <- get_understand_output_path(conf)
+  code_language <- get_understand_code_language(conf)
+  save_path <- arguments[["<save_path>"]]
+  node_file <- paste0(save_path, "/node.csv")
```
node_list

carlosparadis commented on exec/understand.R:
```diff
@@ -0,0 +1,68 @@
+  code_language <- get_understand_code_language(conf)
+  save_path <- arguments[["<save_path>"]]
+  node_file <- paste0(save_path, "/node.csv")
+  edge_file <- paste0(save_path, "/edge.csv")
```
edge_list

carlosparadis commented on R/src.R:
```diff
@@ -4,8 +4,141 @@
+############## Understand Project Builder ##############
+#' Build Understand Analysis Folder
+#'
+#' @description This function builds the data files for Understand from the project_path folder, reading from files that are written in the target language into output_dir
```
remove description tags.

carlosparadis reviewed (changes requested):
Please make sure other PRs you have worked on do not .idea folder and apply the requested changes.

RavenMarQ committed:
Final Push

Standardizing understand.R as well as uploading depends and linemetrics executables. Additional testing required on non-Windows devices needed.

RavenMarQ committed:
Merge branch 'master' into 322-creating-exec-for-notebooks

daomcgill commented on exec/understand.R:
```diff
@@ -0,0 +1,57 @@
+#!/usr/local/bin/Rscript
+
+require(kaiaulu, quietly = TRUE)
```
Review from November

daomcgill reviewed (changes requested):
see comments posted in line for the PR Discussions

daomcgill reviewed (changes requested):
Still has directory creation.

RavenMarQ committed:
Addressing Code Reviews

-Updated Functionality to encapsulate the third function "export" and the new parameter
- Removed "description" tag from function

RavenMarQ committed:
Delete .idea directory

As directed by PR code review

RavenMarQ committed:
Merge branch '322-creating-exec-for-notebooks' of https://github.com/sailuh/kaiaulu into 322-creating-exec-for-notebooks
